### PR TITLE
Upgrade Knative Serving to v0.8.1

### PR DESCRIPTION
**Pull request by rakesh-garimella**
**Description**

Upgrade knative serving v0.8.1
- Copied serving and build yamls into config/crd directory as the yamls added field `versions` which is only supported in k8s 1.15. However kubebuilder as of now supports k8s v .1.14
- Updated knative serving tests to 0.8.1 also upgraded the Gopkg.toml.
- Modified `knative-ingress-gateway` by copying the spec from `kyma-gateway` to make https work

**Related issue(s)**

#5497

### CODE REVIEW

File: components/function-controller/pkg/webhook/default_server/webhook_suite_test.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ func TestMain(m *testing.M) {
 
 	var err error
 	if cfg, err = t.Start(); err != nil {
+		stdlog.Print("its an ERROR!!!")
```
Comment (k15r) [inline components/function-controller/pkg/webhook/default_server/webhook_suite_test.go]: o'rly? what is an error? 
left over debug message?

File: components/function-controller/pkg/utils/serving_utils.go [hunk 2]
```diff
@@ -46,7 +46,7 @@ func GetServiceSpec(imageName string, fn serverlessv1alpha1.Function, rnInfo *Ru
 		Template: &servingv1alpha1.RevisionTemplateSpec{
 			Spec: servingv1alpha1.RevisionSpec{
 				RevisionSpec: v1beta1.RevisionSpec{
-					PodSpec: v1beta1.PodSpec{
+					PodSpec: corev1.PodSpec{
```
Comment (k15r) [inline components/function-controller/pkg/utils/serving_utils.go]: is this really the same? v1beta1 was from githyb.com/knative/serving/apis/serving/v1beta1, now it is from kubernetes-core

File: resources/knative-serving/README.md [hunk 3]
```diff
@@ -12,4 +12,4 @@ Kyma-specific changes:
  * The `config-domain` is made configurable by specifying the `.Values.global.domainName` as the helm template.
  * The `knative-serving` Namespace is no longer created. This happens during the installation process.
  * The image versions are changed to use the release tag.
- * Knative Serving uses the Kyma Istio Ingress gateway.
+ * knative-ingress-gateway has been updated
```
Comment (k15r) [inline resources/knative-serving/README.md]: we might want to point out that the knative-ingress-gateway is now basically a copy of the kyma-gateway, just in a different namespace

File: resources/knative-serving/requirements.yaml [hunk 4]
```diff
@@ -1,3 +1,2 @@
 dependencies:
-  - name: knative-serving-crds-init
-  - name: knative-serving-crds
+  # - name: knative-serving-init
```
Comment (k15r) [inline resources/knative-serving/requirements.yaml]:
```suggestion
```

File: tests/knative-serving/Gopkg.toml [hunk 5]
```diff
@@ -1,11 +1,32 @@
...
 [prune]
   go-tests = true
   unused-packages = true
 
-[[constraint]]
-  name = "github.com/kyma-project/kyma"
-  branch = "master"
+# [[constraint]]
+#   name = "github.com/rakesh-garimella/kyma/common"
+#   branch = "upgrade-knative-081"
```
Comment (k15r) [inline tests/knative-serving/Gopkg.toml]:
```suggestion
```

File: tests/knative-serving/Gopkg.toml [hunk 6]
```diff
@@ -1,11 +1,32 @@
+required = ["github.com/kyma-project/kyma/common"]
 [prune]
   go-tests = true
   unused-packages = true
 
-[[constraint]]
-  name = "github.com/kyma-project/kyma"
-  branch = "master"
+# [[constraint]]
+#   name = "github.com/rakesh-garimella/kyma/common"
```
Comment (k15r) [inline tests/knative-serving/Gopkg.toml]:
```suggestion
```

File: tests/knative-serving/Gopkg.toml [hunk 7]
```diff
@@ -1,11 +1,32 @@
+required = ["github.com/kyma-project/kyma/common"]
 [prune]
   go-tests = true
   unused-packages = true
 
-[[constraint]]
-  name = "github.com/kyma-project/kyma"
-  branch = "master"
+# [[constraint]]
```
Comment (k15r) [inline tests/knative-serving/Gopkg.toml]:
```suggestion
```

File: resources/knative-serving/README.md [hunk 8]
```diff
@@ -5,11 +5,10 @@
...
- * https://github.com/knative/serving/releases/download/v0.6.1/serving.yaml
- * https://github.com/knative/eventing/releases/download/v0.6.1/release.yaml
+ * https://github.com/knative/serving/releases/download/v0.8.1/serving.yaml
 
 Kyma-specific changes:
  * The `config-domain` is made configurable by specifying the `.Values.global.domainName` as the helm template.
  * The `knative-serving` Namespace is no longer created. This happens during the installation process.
  * The image versions are changed to use the release tag.
- * Knative Serving uses the Kyma Istio Ingress gateway.
+ * knative-ingress-gateway is now a copy from kyma-gateway.
```
Comment (bszwarc) [inline resources/knative-serving/README.md]:
```suggestion
 * The `knative-ingress-gateway` is now a copy of `kyma-gateway`.
```

**Comment by antoineco** [comment]
`time="2019-09-19T12:53:04Z" level=fatal msg="while reading dex config from cluster: Get https://10.43.240.1:443/api/v1/namespaces/kyma-system/secrets/admin-user: dial tcp 10.43.240.1:443: connect: connection refused"`
/test pre-master-kyma-gke-upgrade

**Comment by antoineco** [comment]
`(gcloud.dns.record-sets.transaction.execute) HTTPError 412: Precondition not met for 'entity.change.deletions[0]'`
/test pre-master-kyma-gke-integration

**Comment by antoineco** [comment]
GREEN 🎉

**Pull request merged**